src/Common/tests/System/Collections/IEnumerable.Generic.Tests.cs

```diff
@@ -64,6 +64,18 @@ public abstract class IEnumerable_Generic_Tests<T> : TestBase<T>
         protected bool MoveNextAtEndThrowsOnModifiedCollection { get { return true; } }
 
         /// <summary>
+        /// Used in IEnumerable_Generic_Enumerator_Current.
+        /// Some enumerators do not throw accessing Current if enumeration has not yet started (e.g. ConcurrentDictionary)
+        /// </summary>
+        protected virtual bool IEnumerable_Generic_Enumerator_Current_EnumerationNotStarted_ThrowsInvalidOperationException => true;
+
+        /// <summary>
+        /// Some collections (e.g. ConcurrentDictionary) use yield return iterators to enumerate,
+        /// so calling reset anytime throws a NotSupportedException.
+        /// </summary>
+        protected virtual bool IEnumerable_Generic_Enumerator_Reset_Implemented => false;
```

Comment:
It's dead code that I found its way in when I was diagnosing why ResetImplemented also assumes that Reset is implemented or not in the keys or values collections of an IDictionary. Will remove